OrbitSsh/TunnelManager.cpp

```diff
@@ -18,33 +17,22 @@ TunnelManager::TunnelManager(Session* session_ptr, int local_port,
     : socket_("127.0.0.1", local_port),
       channel_(session_ptr, "127.0.0.1", remote_port) {}
 
-ResultType TunnelManager::Connect() {
-  LocalSocketManager::State socket_state = socket_.Tick();
-  DirectTcpIpChannelManager::State channel_state = channel_.Tick();
-
-  if (socket_state == LocalSocketManager::State::kRunning &&
-      channel_state == DirectTcpIpChannelManager::State::kRunning) {
-    return ResultType::kSuccess;
-  }
-  return ResultType::kAgain;
+outcome::result<void> TunnelManager::Connect() {
+  OUTCOME_TRY(socket_.Connect());
+  OUTCOME_TRY(channel_.Initialize());
+  return outcome::success();
 }
 
-ResultType TunnelManager::ReceiveSocketWriteChannel() {
-  std::string text;
-  ResultType result = socket_.Receive(&text);
-
-  if (result != ResultType::kSuccess) return result;
-
-  return channel_.WriteBlocking(text);
+outcome::result<void> TunnelManager::ReceiveSocketWriteChannel() {
+  OUTCOME_TRY(data, socket_.Receive());
+  OUTCOME_TRY(channel_.WriteBlocking(data));
```

Comment:
maybe check in between if data is empty, which means socket is disconnected, which in turn should maybe mean to go into kNotConnected state here. We can also change that later

Reply:
done.